File: PHP/Builder/Vehicle.php
<?php
namespace DesignPatterns\Builder;
class Vehicle 
{
    private  $engine;
    private  $color;
    private  $wheels;

    public function setEngine(String $engine) {
        $this->engine = $engine;
    }
    public function setColor(String $color) {
        $this->color = $color;
    }
    
    public function setWheels(int $wheels) {
        $this->wheels = $wheels;
    }
}

File: Go/builder/Vehicle.go
package builder

import "fmt"

//type Vehicle struct {
//	topSpeed int
//	color    string
//}
type Vehicle struct {
	Wheels    int
	Seats     int
	TopSpeed int
	Structure string
}
type IVehilce interface {
	AcclerateSpeed(speed int) string
}
func (v Vehicle) AcclerateSpeed(speed int) {
fmt.Println("speed of ",v.Structure," is ",speed, "instead of ",v.TopSpeed)
}